controller/rest/remote_export.go

```diff
@@ -0,0 +1,249 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/neuvector/neuvector/controller/api"
+	"github.com/neuvector/neuvector/controller/common"
+	"github.com/neuvector/neuvector/controller/export"
+	"github.com/neuvector/neuvector/share"
+	log "github.com/sirupsen/logrus"
+)
+
+///////////////////////////////////////////////////////////
+// Remote Export Repository REST Handlers /////////////////
+///////////////////////////////////////////////////////////
+
+func handlerRemoteExportRepositoryPost(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	log.WithFields(log.Fields{"URL": r.URL.String()}).Debug()
+	defer r.Body.Close()
+
+	acc, login := getAccessControl(w, r, "")
+	if acc == nil {
+		return
+	} else if !acc.Authorize(&share.CLUSRemoteExportRepository{}, nil) {
+		restRespAccessDenied(w, login)
+		return
+	}
+
+	body, _ := ioutil.ReadAll(r.Body)
+	var remoteExportRepository api.REST_RemoteExportRepository_POST
+	err := json.Unmarshal(body, &remoteExportRepository)
+	if err != nil {
+		msg := fmt.Sprintf("Could not unmarshal request body: %s", err.Error())
+		restRespErrorMessage(w, http.StatusBadRequest, api.RESTErrInvalidRequest, msg)
+		return
+	} else if !isObjectNameValid(remoteExportRepository.Alias) {
```

Comment:
In 5.3 we only allow one entry with alias as "default".
Please check whether the Alias field is "default". Return error if it's not.

Reply:
With the way the endpoints work, only 1 remote repository can be created via the http api with the nickname/alias "default".

However, I just realized that I'll have ensure the object and the path parameter dont have different names.